### ISSUE TITLE
L059 with prefer quoted identifiers wants `IF`

### ISSUE THREAD

**Issue by jasonaowen** [sqlfluff/sqlfluff#2821]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting a `DROP TABLE IF EXISTS...` statement with [L059](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L059)'s prefer quoted identifiers option enabled incorrectly flags the `IF` keyword as needing quotes.

### Expected Behaviour

No error on the `IF` keyword.

### Observed Behaviour

> L:   1 | P:  12 | L059 | Missing quoted identifier IF.

### How to reproduce

With `drop-if.sql` as

```sql
DROP TABLE IF EXISTS example;
```

and `drop-if.cfg` as

```
[sqlfluff:rules:L059]
prefer_quoted_identifiers = True
```

Running SQLFluff will produce an extraneous error. (Note that the second error, about the unquoted `example`, is expected.)

```sh
$ sqlfluff lint --verbose --ignore-local-config --config drop-if.cfg --dialect mysql drop-if.sql 
==== sqlfluff ====
sqlfluff:               0.11.0 python:                 3.8.10
implementation:        cpython dialect:                 mysql
verbosity:                   1 templater:               jinja

==== readout ====

=== [ path: drop-if.sql ] ===

== [drop-if.sql] FAIL                                                                                                                                                    
L:   1 | P:  12 | L059 | Missing quoted identifier IF.
L:   1 | P:  22 | L059 | Missing quoted identifier example.
==== summary ====
violations:        2 status:         FAIL
All Finished 📜 🎉!

$ sqlfluff lint --verbose --ignore-local-config --config drop-if.cfg --dialect postgres drop-if.sql 
==== sqlfluff ====
sqlfluff:               0.11.0 python:                 3.8.10
implementation:        cpython dialect:              postgres
verbosity:                   1 templater:               jinja

==== readout ====

=== [ path: drop-if.sql ] ===

== [drop-if.sql] FAIL                                                                                                                                                    
L:   1 | P:  12 | L059 | Missing quoted identifier IF.
L:   1 | P:  22 | L059 | Missing quoted identifier example.
==== summary ====
violations:        2 status:         FAIL
All Finished 📜 🎉!
```

### Dialect

MySQL, PostgreSQL

### Version

sqlfluff, version 0.11.0
Python 3.8.10

### Configuration

```
[sqlfluff:rules:L059]
prefer_quoted_identifiers = True
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jasonaowen** [comment]
I'm super impressed by the turnaround time on all three of these - thanks again, @tunetheweb and @barrywhart!
